src/storage/cloud_storage/mod.rs

```diff
@@ -207,32 +210,30 @@ impl<U: Sync + Send> StorageBackend<U> for CloudStorage {
     type File = Object;
     type Metadata = ObjectMetadata;
 
-    fn metadata<P: AsRef<Path>>(&self, _user: &Option<U>, path: P) -> Box<dyn Future<Item = Self::Metadata, Error = Error> + Send> {
-        let uri = match self.uris.metadata(path) {
-            Ok(uri) => uri,
-            Err(err) => return Box::new(future::err(err)),
-        };
+    async fn metadata<P: AsRef<Path> + Send>(&self, _user: &Option<U>, path: P) -> Result<Self::Metadata, Error> {
+        let uri: Uri = self.uris.metadata(path)?;
 
-        let client = self.client.clone();
+        let client: Client<HttpsConnector<HttpConnector<GaiResolver>>, Body> = self.client.clone();
 
-        let result = self
-            .get_token()
-            .and_then(|token| {
-                Request::builder()
-                    .uri(uri)
-                    .header(header::AUTHORIZATION, format!("{} {}", token.token_type, token.access_token))
-                    .method(Method::GET)
-                    .body(Body::empty())
-                    .map_err(|_| Error::from(ErrorKind::PermanentFileNotAvailable))
-            })
-            .and_then(move |request| client.request(request).map_err(|_| Error::from(ErrorKind::PermanentFileNotAvailable)))
-            .and_then(unpack_response)
-            .and_then(|body_string| {
-                serde_json::from_slice::<Item>(&body_string)
-                    .map_err(|_| Error::from(ErrorKind::PermanentFileNotAvailable))
-                    .and_then(|item| item_to_metadata(&item))
-            });
-        Box::new(result)
+        let token: Token = self.get_token().compat().await?;
```

Comment:
I think its actually nice that you declare the types explicitly, I'm gonna see if I can also do that. It will probably make things clearer.

Reply:
I think that the rust-analyzer plugin (at least for Visual Studio Code) provides this as a quick fix